src/zeit/cms/content/reference.py

```diff
@@ -45,6 +62,15 @@ def __get__(self, instance, class_):
             xml_reference_name=self.xml_reference_name)
 
     def __set__(self, instance, value):
+        """Writes a ``value`` into XML of ``instance``.
+
+        The items in ``value`` should either provide ``IReference``` or should
+        be adaptable to ``IXMLReference``. In the second case each item is
+        adapted to ``IXMLReference`` and the result is wrapped into an
+        ``IReference`` via ``_ensure_references``.
+
+        """
+        value = self._ensure_references(instance, value)
```

Comment:
Meiner Ansicht nach darf das nicht in die ReferenceProperty, weil man damit zum Datenverlust einlädt:
Existierende Referenzen können ja eigene Daten haben (z.B. bei Autoren die location). Wenn die API ist "schreibe Referenz-Objekte drauf", dann ist Clients (hoffentlich ;) bewusst, dass man beim Schreiben auch die location mit setzt.
Wenn die API aber ist "du kannst mir auch nur ICMSContent geben", dann versteckt das, dass es da noch Zusatzdaten geben könnte (die beim Schreiben dann stillschweigend plattgemacht werden). Das finde ich keine gute Idee.

Insofern sollte das denke ich in eine separate abgeleitete Property (die kann gerne auch in z.c.video leben), wo man dick dranschreibt, dass das Absicht ist mit dem "Datenverlust" (weil er bei Videos eh nicht auftreten kann, weil's location dort nie gibt).